app/AppContainer.js

```diff
@@ -110,5 +115,14 @@ export class AppContainer extends Component {
         this.setCaughtException(e);
         XH.handleException(e, {requireReload: true});
     }
+
+
+    //------------------------
+    // Implementation
+    //------------------------
+    unauthorizedMessage() {
+        const role = XH.appModel.requireRole;
+        return span(`Permission "${role}" is required to use this application.`, <br />, 'Please request this permission from the Help Desk.');
```

Comment:
I don't think we need to bake in concept of Help Desk at this level - apps might well have their own support list or procedure.  Let's leave that off and if we need support for a more customized message in the future we can decide how we wish to implement.